Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs

```diff
@@ -36,125 +41,348 @@ public class CLIProcessor
+
+            Reporter.ToLog(eLogLevel.DEBUG, string.Format("########################## Starting Automatic {0} Execution Process ##########################", GingerDicser.GetTermResValue(eTermResKey.RunSet)));
+            Reporter.ToLog(eLogLevel.DEBUG, string.Format("Parsing {0} execution arguments...", GingerDicser.GetTermResValue(eTermResKey.RunSet)));
+            Reporter.ToLog(eLogLevel.INFO, $"Solution: {runOptions.Solution}");
+            Reporter.ToLog(eLogLevel.INFO, $"Runset: {runOptions.Runset}");
+            Reporter.ToLog(eLogLevel.INFO, $"Environment: {runOptions.Environment}");
+            Reporter.ToLog(eLogLevel.DEBUG, "Loading Configurations...");
+            
+            mCLIHandler = new CLIArgs();
+            mCLIHelper.Solution = runOptions.Solution;
+            mCLIHelper.Runset = runOptions.Runset;
+            mCLIHelper.Env = runOptions.Environment;
+            mCLIHelper.RunAnalyzer = runOptions.RunAnalyzer;
+            mCLIHelper.ShowAutoRunWindow = runOptions.ShowAutoRunWindow;
+            mCLIHelper.TestArtifactsFolder = runOptions.TestArtifactsPath;
 
-                mCLIHelper.SetTestArtifactsFolder();                
+            ExecuteRunSet();
+
+            return Environment.ExitCode;
+        }
+
+        private void SetVerboseLevel(OptionsBase.eVerboseLevel verboseLevel)
+        {            
+            if (verboseLevel == OptionsBase.eVerboseLevel.debug)
+            {
+                Reporter.AppLoggingLevel = eAppReporterLoggingLevel.Debug;
             }
+            else
+            {
+                Reporter.AppLoggingLevel = eAppReporterLoggingLevel.Normal;
+            }
+            
+        }
 
-            Reporter.ToLog(eLogLevel.DEBUG, string.Format("Executing..."));
-            Execute();
+        private int HandleCLIParseError(IEnumerable<Error> errs)
+        {
+            if (errs.Count() == 1 && errs.First() is HelpVerbRequestedError)
+            {
+                // User requested help 
+                PrintGingerCLIHelp();
+                return 0;
+            }
+            else if (errs.Count() == 1 && errs.First() is VersionRequestedError)
+            {
+                // User requested version
+                PrintGingerVersionInfo();
+                return 0;
+            }
+            else
+            {
+                Reporter.ToConsole(eLogLevel.ERROR, "Please fix the arguments and try again");
+                return 1;
+            }
+            
+        }
 
-            Reporter.ToLog(eLogLevel.DEBUG, "Closing Solution and doing Cleanup...");
-            mCLIHelper.CloseSolution();
+        private void PrintGingerVersionInfo()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.Append(Environment.NewLine);
+            stringBuilder.Append("Ginger Executor: ").Append(Assembly.GetEntryAssembly().Location).Append(Environment.NewLine);            
```

Comment:
why not to get version from ApplicationInfo.ApplicationVersion

Reply:
it show which exe/dll is running not the version